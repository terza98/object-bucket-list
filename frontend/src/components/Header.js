import React from 'react';
import Navbar from 'react-bootstrap/Navbar'

export default function Header(props){
    return(
        <>
            <Navbar bg="dark" variant="dark">
                <Navbar.Brand href="#home">
                    {props.title}
                </Navbar.Brand>
            </Navbar>
        </>
    );
}