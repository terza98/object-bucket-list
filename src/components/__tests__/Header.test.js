import React from 'react';
import Header from '../Header.js';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';

let container = null;
beforeEach(() => {
	// setup a DOM element as a render target
	container = document.createElement('div');
	document.body.appendChild(container);
});

afterEach(() => {
	// cleanup on exiting
	unmountComponentAtNode(container);
	container.remove();
	container = null;
});

it('renders without crashing', () => {
	ReactDOM.render(<Header />, container);
});

it('matches snapshot', () => {
	const { asFragment } = render(<Header />);

	expect(asFragment()).toMatchSnapshot();
});

test('renders simple title', () => {
	const title = 'Example Title';
	const { getByText } = render(<Header title={title} />);
	const headerText = getByText(title);
	expect(headerText).toBeInTheDocument();
});
