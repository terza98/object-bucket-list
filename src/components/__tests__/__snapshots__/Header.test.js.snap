// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
    id="navbar"
  >
    <a
      class="navbar-brand"
      href="/"
    />
  </nav>
</DocumentFragment>
`;
